### PR TITLE
Add benchmarks for array access, create and pop operations.

### LINKED ISSUE #428: [Feature Request] - Create benchmark for Array manipulation
In order to test the upcoming [optimisation](https://github.com/boa-dev/boa/pull/419) work we're doing around objects it would be good to have a benchmark for Arrays and manipulating them.

## Example
Benchmarks live [here](https://github.com/boa-dev/boa/tree/master/boa/benches)
an array creation and access would be categorised under `exec`.

If exec benchmarks become too big we can compartmentalise them later on.

You can see the fibonacci benchmark here:
https://github.com/boa-dev/boa/blob/master/boa/benches/exec.rs#L49-L66

2 benchmarks could be made, 1 for accessing a value from an array, another for creating an array.
We could introduce more benchmarks here for various methods commonly used on arrays

in order to run the benchmarks you can run `cargo bench` in the root folder

## Contributing
https://github.com/boa-dev/boa/blob/master/CONTRIBUTING.md
https://github.com/boa-dev/boa/blob/master/docs/debugging.md

### PR DESCRIPTION
**Pull request by abhijeetbhagat**
This Pull Request fixes/closes #428 .

It adds benchmarks for array access, create and pop operations.

### CODE REVIEW

**Comment by abhijeetbhagat** [comment]
I am not sure why the older commits are included in this PR. I even tried changing the base to a different branch to force GH to only include the latest commit.

**Comment by abhijeetbhagat** [comment]
@jasonwilliams review?

**Comment by Razican** [comment]
> I am not sure why the older commits are included in this PR. I even tried changing the base to a different branch to force GH to only include the latest commit.

No worries, we will squash + merge :)

> @jasonwilliams review?

This depends partly on #431. But this can already be adapted to that. The idea is to use the syntax in that PR to only benchmark the execution, without benchmarking the lexing, parsing and realm creation.

**Comment by abhijeetbhagat** [comment]
Understood. I'll take the lexer out of the benchmarking equation by following the changes in the linked issue.

**Comment by Razican** [comment]
Let us know, @abhijeetbhagat, when this is ready :)

File: boa/benches/exec.rs [hunk 1]
```diff
@@ -291,12 +291,89 @@ fn regexp(c: &mut Criterion) {
...
+    let nodes = Parser::new(&black_box(lexer.tokens)).parse_all().unwrap();
+
+    c.bench_function("Array creation (Execution)", move |b| {
+        b.iter(|| black_box(&nodes).run(&mut engine).unwrap())
+    });
+}
+
+static ARRAY_POP: &str = r#"
+(function(){
+    let testArray = [83, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36, 28];
```
Comment (HalidOdat) [inline boa/benches/exec.rs:345]: Maybe we should break the array in multiple lines, so its not in on line.

**Comment by HalidOdat** [comment]
The benchmarks are failing because of `criterion-compate-action` github action its fixed on master, rebasing should fix it :)

File: boa/benches/exec.rs (hunk 1 above)
Comment (abhijeetbhagat) [reply boa/benches/exec.rs]: Ok let me fix that.

**Comment by abhijeetbhagat** [comment]
I thought `pop()` was causing a problem. So i removed it in the latest push. Let me restore it and also fix the array formatting.

**Pull request merged**